### DISCUSSION TITLE
The number of subfolders doesn't match the number of specified classes when using custom Dataset

### DISCUSSION THREAD

**Discussion by foreverls** [open-mmlab/mmclassification#1341] [Q&A] [answered]
I am following the tutorial here to use my own dataset for finetuning.
https://mmclassification.readthedocs.io/en/latest/tutorials/new_dataset.html#an-example-of-customized-dataset

The dataset structure is similar to below, but only two classes(Female/Male). and same for test and val folder.
<img width="505" alt="image" src="https://user-images.githubusercontent.com/48796858/216017383-ba371880-29a7-4bd4-bc55-3bd0be89f50a.png">

the config looks like
```
dataset_type = 'CustomDataset'
classes = ['Female', 'Male']  # The category names of your dataset

data = dict(
    train=dict(
        type=dataset_type,
        data_prefix='data/gender_dataset/train',
        classes=classes,
        pipeline=train_pipeline
    ),
    val=dict(
        type=dataset_type,
        data_prefix='data/gender_dataset/val',
        classes=classes,
        pipeline=test_pipeline
    ),
    test=dict(
        type=dataset_type,
        data_prefix='data/gender_dataset/test',
        classes=classes,
        pipeline=test_pipeline
    )
)
```
but when I train it, it will show error: AssertionError: The number of subfolders (3) doesn't match the number of specified classes (2). Please check the data folder.
May I know what is wrong with the folder structure?  Shall I also create an annotation file even with such folder structure? thanks a lot!

**Comment by Ezra-Yu** [comment, marked as answer]
> The dataset structure is similar to below, but only two classes(Female/Male). and same for test and val folder.

Check there are only 2 sub-folders, both in your train/val/test dataset root.

**Reply by foreverls** [reply]
thanks for replying!  here is my folder structure.
![image](https://user-images.githubusercontent.com/48796858/216224072-e4b74cda-f98b-4249-bc07-2ef2b1f3b23f.png)

**Reply by foreverls** [reply]
I guess I know why...there is a hidden folder inside...Thanks!
